app/app.py

```diff
@@ -0,0 +1,251 @@
+# thoth-document-sync
+# Copyright(C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Sync Thoth documents to an S3 API compatible remote."""
+
+from __future__ import annotations
+
+import logging
+import os
+import itertools
+import re
+from concurrent.futures.thread import ThreadPoolExecutor
+from concurrent.futures import as_completed
+from boto3 import client
+from botocore.exceptions import ClientError
+
+from enum import Enum
+from collections import Counter, defaultdict
+from typing import cast, Optional, Iterable, Tuple, Type, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.client import S3Client
+
+import click
+from thoth.common import init_logging
+from thoth.storages import __version__ as __thoth_storages_version__
+from thoth.storages import AnalysisByDigest
+from thoth.storages import AnalysisResultsStore
+from thoth.storages import SolverResultsStore
+from thoth.storages import CephStore
+from thoth.storages.result_base import ResultStorageBase
+
+from prometheus_client import CollectorRegistry, Gauge, Counter as PromCounter, push_to_gateway
+from .lazy_set_ops import sorted_iter_set_difference
+
+prometheus_registry = CollectorRegistry()
+
+__version__ = "0.3.1"
+__component_version__ = f"{__version__}+storages{__thoth_storages_version__}"
+
+init_logging()
+_LOGGER = logging.getLogger("thoth.document_sync")
+
+_DEFAULT_CONCURRENCY = 16
```

Comment:
Lets add this as well with env vars, so we can adjust this from deployment without having to change the source code.
```suggestion
_DEFAULT_CONCURRENCY = os.getenv("THOTH_DEFAULT_CONCURRENCY",16)
```

Reply:
That's handled by the `--concurrency` flag (with click) with a var env as a default.
(I didn't added this, but seems like github is not great for file renames tracking...)